examples/huggingface-transformers/benchmark.py

```diff
@@ -0,0 +1,394 @@
+    input_names = []
+    if args.engine in [DEEPSPARSE_ENGINE, ORT_ENGINE]:
+        args.model_filepath, input_names, _ = overwrite_transformer_onnx_model_inputs(
+            args.model_filepath,
+            batch_size=args.batch_size,
+            max_length=args.max_sequence_length,
+        )
+
+    # load model
+    if args.engine == DEEPSPARSE_ENGINE:
+        print(f"Compiling deepsparse model for {args.model_filepath}")
+        model = compile_model(
+            args.model_filepath, args.batch_size, args.num_cores, args.num_sockets
+        )
+    elif args.engine == ORT_ENGINE:
+        print(f"loading onnxruntime model for {args.model_filepath}")
+
+        sess_options = onnxruntime.SessionOptions()
+        if args.num_cores is not None:
+            sess_options.intra_op_num_threads = args.num_cores
+        sess_options.log_severity_level = 3
+        sess_options.graph_optimization_level = (
+            onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
+        )
+        model = onnxruntime.InferenceSession(
+            args.model_filepath, sess_options=sess_options
+        )
+
+    return model, input_names
+
+
+def _load_data(args, input_names) -> List[List[numpy.ndarray]]:
+    if args.data_path.startswith("zoo:"):
+        data_dir = Zoo.load_model_from_stub(
+            args.data_path
+        ).data_inputs.downloaded_path()
+    else:
+        data_dir = args.data_path
+        data_files = os.listdir(data_dir)
+        if any(".npz" not in file_name for file_name in data_files):
+            raise RuntimeError(
+                f"All files in data directory {data_dir} must have a .npz extension "
+                f"found {[name for name in data_files if '.npz' not in name]}"
+            )
+
+    samples = load_numpy_list(data_dir)
+    processed_samples = []
+    warning_given = False
+    for sample in samples:
+        if not all(inp_name in sample for inp_name in input_names) or len(
+            input_names
+        ) != len(sample):
+            if not warning_given:
+                warnings.warn(
+                    "input sample found whose input names do not match the model input "
+                    "names, this may cause an exception during benchmarking"
+                )
+                warning_given = True
+            sample = list(sample.values())
+        else:
+            sample = [sample[inp_name] for inp_name in input_names]
+
+        for idx, array in enumerate(sample):
+            processed_array = numpy.zeros(
+                [args.max_sequence_length, *array.shape[1:]],
+                dtype=array.dtype,
+            )
+            if array.shape[0] < args.max_sequence_length:
+                processed_array[: array.shape[0], ...] = array
+            else:
+                processed_array[:, ...] = array[: args.max_sequence_length, ...]
+            sample[idx] = processed_array
+        processed_samples.append(sample)
+    return processed_samples
+
+
+def _iter_batches(
```

Comment:
For the future would be good to see how we might merge this logic into the sparsezoo dataset class so we can reuse it and make the integration with the sparsezoo more seemless

Reply:
definitely, looks like adding an `iter_batches` function to `Dataset` should work out fine. will add to the backlog

Reply:
agreed. It makes sense that there should be some common processing method for datasets. Otherwise any gains may be attributed to model and methods vs differences in testing